### PR TITLE
Update readme and issue template.

### PR DESCRIPTION
**Pull request by Unnoen**
Updated README.md to make it prettier and read nicer.

Added forge logo, description, some badges, adjusted titles for better clarity, mention of patreon in the footer, also added a "common links" index.

I also added a "support table", the goal being a clear-cut list of what is supported and also to stop "X version when?"

The issue template has been adjusted to include common placeholders.

All the .md documents have been moved to `/docs` I'm not sure of the stance of this so let me know what layout is wanted, but github will automatically pull docs from there (like displaying readme in repo root) so functionally it's the same.

### CODE REVIEW

**Comment by Unnoen** [comment]
The latest version badge pulls from Forge's maven-metadata.xml so will *not* need to be updated on every build.

The support table would have three types, Development, Maintained and Legacy. I was going to add a key to the table to show what those mean but I feel it's unnecessary, at least at the moment. This *will* need to be manually adjusted, but since it's major versions, it will be very rare.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+* [Forum]
+* [Discord]
+* [Documentation]
+
+#### Notes:
+- Introduced in 1.13 was a new FML, information found [here](NewFML.md).
+
+# Installing Forge
+
+Go to [the Forge website](http://files.minecraftforge.net)
```
Comment (bs2609) [inline docs/README.md]: `https` links wherever possible would be nice.

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ and select the minecraft version you wish to get forge for from the list.
```
Comment (bs2609) [inline docs/README.md]: Use 'Minecraft', 'Forge', 'GitHub', etc. throughout, as these are the 'official' names.

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -1,4 +1,16 @@
 <!-- Thank you for reporting! -->
-
 <!-- Please go to the forum first for support with crashes and issues. -->
 <!-- http://www.minecraftforge.net/forum/forum/18-support-bug-reports/ -->
+
+**Minecraft Version:** {mc version}
+
+**Forge Version:** {forge version}
```
Comment (bs2609) [inline .github/ISSUE_TEMPLATE.md]: You might need a "**not** the Minecraft version, **DO NOT** specify 'latest'/'recommended'" here.

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -1,4 +1,16 @@
...
+
+**Full Log:** {link to gist/pastebin with full latest.log}
```
Comment (bs2609) [inline .github/ISSUE_TEMPLATE.md]: For GitHub issues, Gist is probably preferred, as it allows larger file sizes and I think stays valid for longer (the poster can be assumed to have the required account, for obvious reasons).

**Review by bs2609 — COMMENTED** [review]
Couple of style points and suggestions:

File: .github/ISSUE_TEMPLATE.md (hunk 3 above)
Comment (Unnoen) [reply .github/ISSUE_TEMPLATE.md]: I don't think it's necessary to mention *not* the Minecraft version as it's mentioned in it's own placeholder directly above.

**Review by 0xilly — COMMENTED** [review]
lgtm maybe add a note that maintained means bug fixes only.

**Comment by marvin-roesch** [comment]
GitHub supports [multiple issue templates](https://help.github.com/en/articles/about-issue-and-pull-request-templates). I think it's a good idea to provide at least two templates as part of this PR: one for bug reports and one for feature requests (which are rare, but do happen).

A nice bonus of multiple templates is that we get to display a short description of what constitutes an issue fitting the template. This might help reduce the occurrence of issues that really belong in the forums, if the description is properly worded.

**Comment by Unnoen** [comment]
> lgtm maybe add a note that maintained means bug fixes only.

"Maintained" was to mean "active". Once 1.14 dropped 1.12 would become "Legacy". If there's a better suggestion for the statuses let me know, but I was suggesting as follows:

**Development**: Work being done in private/Patches (No public build)
**Maintained**: Fully supported and actively receiving work.
**Legacy**: Limited support. Bug fixes only.

**Comment by Unnoen** [comment]
> GitHub supports [multiple issue templates](https://help.github.com/en/articles/about-issue-and-pull-request-templates). I think it's a good idea to provide at least two templates as part of this PR: one for bug reports and one for feature requests (which are rare, but do happen).
> 
> A nice bonus of multiple templates is that we get to display a short description of what constitutes an issue fitting the template. This might help reduce the occurrence of issues that really belong in the forums, if the description is properly worded.

Ah yes, I saw that there was a new issue template method but didn't look into it, it does seem very useful. I'll make some for bugs and features and look through existing issues to see if there's anything else that might deserve it's own template.

**Comment by Unnoen** [comment]
The templates have been added but I'm not sure what you want the content of them to be.

They will auto assign the Bug and Feature label respectively.

At the moment the feature request one is GitHub's default.

**Comment by mezz** [comment]
Thanks for doing this, it looks nice!

Some things to note for your new issue templates though:

- The majority of player bug reports should go to the Forge Forums first. The bug tracker is mainly for developers who have narrowed down the source of the bug so they are confident it comes from Forge itself. Using the forums as the front-line for bug reports helps us cut down on "my game crashed" vague reports here, the participants in the forum thread can help narrow down the bug and figure out what's wrong.

- Feature suggestions are rarely implemented. Generally if someone wants a feature, they should be ready to implement it themselves in a PR and we can work from there. If they are not certain about the details, they can open a Draft PR or discuss it in the Discord before starting on it.

With that in mind, please re-think the issue templates to help give reporters the right expectations.

**Comment by Unnoen** [comment]
Updated the bug report template to mention the forum for support first, the feature request template now mentions that opening a PR/discussion is the best course of action.

I decided to keep the feature request template as it can act as a "notice" for users who decide to open an issue for something they want implemented. Also, it does have its place, for example https://github.com/MinecraftForge/MinecraftForge/issues/5732 as a placeholder/update location.

**Comment by Unnoen** [comment]
*Please note that there are comments in said templates that will only show when viewing raw or in an issue.*

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+about: Visit the forum first for issues with crashes.
+title: ''
+labels: Bug
+assignees: ''
+
+---
+
+<!-- Thank you for reporting! -->
+
+<!-- Please go to the forum first for support with crashes and issues. -->
```
Comment (mezz) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
<!-- For support with crashes and issues, please create a post on the forum instead: -->
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---
+name: Feature request
+about: Want to request a feature?
+title: ''
+labels: Feature
+assignees: ''
+
+---
+
+<!-- Please note, feature requests are rarely implemented. -->
```
Comment (mezz) [inline .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
<!-- Please note, feature requests that are submitted as GitHub issues here are rarely implemented. -->
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+<!-- If you'd like to see something in Forge, you should be ready to implement it yourself. -->
```
Comment (mezz) [inline .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
<!-- If you'd like to see something in Forge, you should be ready to implement it as a PR. -->
```

**Review by mezz — CHANGES_REQUESTED** [review]
Looks good. I'd like to improve a few of the messages to be more direct or slightly more friendly sounding so people who really want to contribute are not turned away.

**Comment by Unnoen** [comment]
Commited the changes proposed and I changed the PR to the 1.14.x branch.

File: docs/README.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+MinecraftForge [![Latest Release](https://img.shields.io/maven-metadata/v/https/files.minecraftforge.net/maven/net/minecraftforge/forge/maven-metadata.xml.svg?label=latest)](https://files.minecraftforge.net) [![Discord](https://img.shields.io/discord/313125603924639766.svg?color=%237289da&label=discord&logo=discord&logoColor=%237289da)](https://discord.gg/UvedJ9m) [![Support](https://img.shields.io/badge/patreon-support-orange.svg)](https://www.patreon.com/LexManos) [![Merch](https://img.shields.io/badge/flocker-merch-blueviolet.svg)](https://forgedev.flocker.tv/)
+=============
+
+Forge is a free, open-source modding API all of your favourite mods use!
+
+| Version  | Support |
+| ------------- | ------------- |
+| 1.14.2  | Maintained  |
+| 1.12.2  | Legacy  |
```
Comment (tterrag1098) [inline docs/README.md]: 1.12.2 will not be in legacy support until 1.14 has a recommended build.

**Review by tterrag1098 — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 8 above)
Comment (Unnoen) [reply docs/README.md]: Oh whoops, I got ahead of myself. 1.13.2 is no longer being maintained though is it?

Comment (Unnoen) [reply docs/README.md]: Also, should I keep "Maintained"? Maybe "Active (latest)" and "Active (stable)"?

**Comment by Unnoen** [comment]
Removed the "legacy" from 1.12.2 and used "Active (latest)" and "Active (stable)" respectively.

**Review by mezz — APPROVED** [review]

**Pull request merged**